protocols/mdns/src/behaviour/socket.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use std::{
+    io::Error,
+    net::{SocketAddr, UdpSocket},
+    task::{Context, Poll},
+};
+
+/// Interface that must be implemented by the different runtimes to use the UdpSocket in async mode
+pub trait AsyncSocket: Send + 'static {
+    /// Create the async socket from the ```std::net::UdpSocket```
+    fn from_socket(socket: UdpSocket) -> std::io::Result<Self>
```

Comment:
```suggestion
    fn from_std(socket: UdpSocket) -> std::io::Result<Self>
```

Just an idea, not attached to it.

Reply:
@thomaseizinger I'll check it but, but the Tokio socket implementation have a function with this name, and might cause conflict

Reply:
If we only import the `AsyncSocket` type at the usage site, it should be unambiguous.